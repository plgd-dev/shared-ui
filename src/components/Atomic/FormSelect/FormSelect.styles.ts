import { css } from '@emotion/react'

import { FormSelectSizeType } from './FormSelect.types'
import { ThemeType, get } from '../_theme'

export const select = (theme: ThemeType, size: FormSelectSizeType, disabled?: boolean) => css`
    .select__control {
        .select__value-container {
            .select__placeholder,
            .select__input-container {
                font-family: 'Poppins', sans-serif;
                font-style: normal;
                font-weight: 400;
                font-size: 14px;
                line-height: 22px;
                color: ${get(theme, `FormSelect.input.color`)}!important;

                input {
                    font-family: 'Poppins', sans-serif !important;
                    font-style: normal !important;
                    font-weight: 400 !important;
                    font-size: 14px !important;
                    line-height: 22px !important;
                    color: ${get(theme, `FormSelect.input.color`)}!important;
                }
            }
        }

        &.select__control--menu-is-open {
            ${!disabled &&
            css`
                border: 1px solid ${get(theme, `FormSelect.menu.open.borderColor`)};
            `}
        }

        &.select__control--is-focused {
            box-shadow: none;
        }
    }

    .select__indicator-separator {
        display: none;
    }

    .select__input-container {
        margin: 0;
    }
`

export const control = (theme: ThemeType) => css`
    height: 44px;
    min-height: 44px;
    border-radius: 8px;
    box-shadow: none;
    transition: all 0.3s;
    border: 1px solid ${get(theme, `FormSelect.control.borderColor`)};
    background: ${get(theme, `FormSelect.control.background`)};

    &:focus {
        outline: none;
    }

    &:hover {
        outline: none;
        border: 1px solid ${get(theme, `FormSelect.control.hover.borderColor`)};
    }
`

export const inlineStyle = (theme: ThemeType) => css`
    border-color: transparent;

    &:hover {
        border-color: ${get(theme, `FormSelect.control.borderColor`)};
    }
`

export const small = css`
    height: 36px;
    min-height: 36px;
`

export const error = (theme: ThemeType) => css`
    border-color: ${get(theme, `FormSelect.error.color`)} !important;
`

export const menu = (theme: ThemeType) => css`
    width: auto !important;
    margin-top: 4px;
    margin-bottom: 4px;
    border: 1px solid ${get(theme, `FormSelect.menu.borderColor`)};
    box-shadow: 0 30px 40px rgba(28, 52, 99, 0.1);
    border-radius: 8px;
    background: ${get(theme, `FormSelect.menu.background`)};
`

export const menuList = css`
    text-align: left;
    padding: 8px 0;
`

export const option = (theme: ThemeType) => css`
    color: ${get(theme, `FormSelect.option.color`)};
    font-style: normal;
    font-weight: 400;
    line-height: 22px;
    font-size: 14px;
    display: flex;
    align-items: center;
    background: transparent;
    transition: all 0.3s;
    cursor: pointer;
    padding: 8px 16px;

    &:hover {
        color: ${get(theme, `FormSelect.option.hover.color`)};
        background: transparent;
    }

    &:active {
        background: transparent;
    }
`

export const optionRight = css`
    justify-content: flex-end;
`

export const optionSelected = (theme: ThemeType) => css`
    color: ${get(theme, `FormSelect.optionSelected.color`)};
    background: transparent;
`

export const selectContainer = css`
    width: 100%;
`

export const selectAutoWidthContainer = css`
    display: inline-flex;
`

export const valueContainer = css`
    padding: 0 0 0 15px; // -1 for border
`

export const value = (theme: ThemeType) => css`
    color: ${get(theme, `FormSelect.value.color`)};
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    line-height: 22px;
    text-align: left;
`

export const dropdownIndicator = css`
    //padding-right: 16px;
    padding: 4px 16px 4px 4px;
`

export const indicator = (theme: ThemeType) => css`
    transition: all 0.3s;
    color: ${get(theme, `FormSelect.indicator.color`)};
`

export const indicatorOpen = css`
    transform: rotate(180deg);
`

export const placeholder = (theme: ThemeType) => css`
    color: ${get(theme, `FormSelect.placeholder.color`)};
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    line-height: 22px;
`

export const textRight = css`
    text-align: right;
`

export const disabled = (theme: ThemeType) => css`
    color: ${get(theme, `FormSelect.disabled.color`)};
`

export const input = css`
    display: flex;
    justify-content: flex-end;

    input {
        text-align: right;
        width: 100% !important;
    }

    &:after {
        display: none !important;
        content: '' !important;
    }
`
