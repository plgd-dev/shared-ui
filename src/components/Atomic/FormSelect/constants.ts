export const selectSizes = {
    NORMAL: 'normal',
    SMALL: 'small',
} as const

export const selectAligns = {
    LEFT: 'left',
    RIGHT: 'right',
} as const
