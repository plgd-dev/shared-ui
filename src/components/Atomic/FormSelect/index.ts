export { default } from './FormSelect'
export * from './FormSelect'

export * from './constants'
