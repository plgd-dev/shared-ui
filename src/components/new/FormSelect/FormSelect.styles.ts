import { css } from '@emotion/react'
import { colors } from '../_utils/colors'
import { fontPrimary } from '../_utils/commonStyles'

export const select = css`
    .select__control {
        height: 44px;
        min-height: 44px;
        border: 1px solid ${colors.neutral300};
        border-radius: 8px;

        .select__value-container {
            .select__placeholder {
                color: #81868c;
            }

            .select__placeholder,
            .select__input-container {
                font-family: 'Poppins', sans-serif;
                font-style: normal;
                font-weight: 400;
                font-size: 14px;
                line-height: 22px;

                input {
                    font-family: 'Poppins', sans-serif !important;
                    font-style: normal !important;
                    font-weight: 400 !important;
                    font-size: 14px !important;
                    line-height: 22px !important;
                    color: ${colors.neutral800}!important;
                }
            }
        }
    }

    .select__single-value {
        text-align: left;
    }

    .select__indicator-separator {
        display: none;
    }
`

export const menu = css`
    margin-top: 4px;
    margin-bottom: 4px;
    border: 1px solid ${colors.neutral300};
    box-shadow: none;
    border-radius: 8px;
`

export const menuList = css`
    text-align: left;
    padding: 8px 0;
`

export const option = css`
    color: ${colors.neutral500};
    font-family: ${fontPrimary};
    font-style: normal;
    font-weight: 600;
    font-size: 14px;
    line-height: 14px;
    display: flex;
    align-items: center;
    background: transparent;
    transition: all 0.3s;
    cursor: pointer;
    padding: 8px 16px;

    &:hover {
        color: ${colors.primaryBonus};
        background: transparent;
    }

    &:active {
        background: transparent;
    }
`

export const optionSelected = css`
    color: ${colors.primaryBonus};
    background: transparent;
`

export const selectContainer = css`
    width: 100%;
`

export const valueContainer = css`
    padding: 0 16px;
`

export const value = css`
    color: ${colors.neutral800};
    font-family: ${fontPrimary};
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    line-height: 22px;
`

export const dropdownIndicator = css`
    padding-right: 16px;
`

export const indicator = css`
    transition: all 0.3s;
    color: ${colors.neutral500};
`

export const indicatorOpen = css`
    transform: rotate(180deg);
`
