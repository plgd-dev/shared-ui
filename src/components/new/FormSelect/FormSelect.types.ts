export type Option = { value: string; label: string }

export type Props = {
    className?: string
    defaultValue?: any
    disabled?: boolean
    isSearchable?: boolean
    name?: string
    options: Option[]
    onChange?: (v: Option) => void
    value?: Option
}
