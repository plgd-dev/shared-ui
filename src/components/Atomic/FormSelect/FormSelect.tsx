import { forwardRef } from 'react'
import Select, {
    components,
    ContainerProps,
    ControlProps,
    DropdownIndicatorProps,
    MenuListProps,
    MenuProps,
    OptionProps,
    PlaceholderProps,
    SingleValueProps,
    ValueContainerProps,
} from 'react-select'

import { GroupedOption, Option, Props, defaultProps } from './FormSelect.types'
import * as styles from './FormSelect.styles'
import { convertSize, IconTableArrowDown } from '../Icon'
import { selectSizes } from './constants'

const FormSelect = forwardRef<any, Props>((props, ref) => {
    const { className, defaultValue, error, disabled, inlineStyle, isSearchable, options, name, menuIsOpen, onChange, size, value, ...rest } = {
        ...defaultProps,
        ...props,
    }
    const stylesOverride = {
        menu: (base: any) => ({
            ...base,
            width: 'max-content',
            minWidth: '100%',
        }),
    }

    const Control = ({ children, ...props }: ControlProps<Option>) => (
        <components.Control
            {...props}
            css={[styles.control, error && styles.error, size === selectSizes.SMALL && styles.small, inlineStyle && styles.inlineStyle]}
        >
            {children}
        </components.Control>
    )

    const ValueContainer = ({ children, ...props }: ValueContainerProps<Option>) => (
        <components.ValueContainer {...props} css={styles.valueContainer}>
            {children}
        </components.ValueContainer>
    )

    const SelectContainer = ({ children, ...props }: ContainerProps<Option>) => (
        <components.SelectContainer {...props} css={styles.selectContainer}>
            {children}
        </components.SelectContainer>
    )

    const SingleValue = ({ children, ...props }: SingleValueProps<Option>) => (
        <components.SingleValue {...props} css={[styles.value, props.selectProps.isDisabled && styles.disabled]}>
            {children}
        </components.SingleValue>
    )

    const DropdownIndicator = (props: DropdownIndicatorProps) => (
        <components.DropdownIndicator {...props} css={styles.dropdownIndicator}>
            <div css={[styles.indicator, props.selectProps.menuIsOpen && styles.indicatorOpen]}>
                <IconTableArrowDown {...convertSize(10)} />
            </div>
        </components.DropdownIndicator>
    )

    const Menu = (props: MenuProps<Option, false, GroupedOption>) => (
        <components.Menu<Option, false, GroupedOption> {...props} css={styles.menu}>
            {props.children}
        </components.Menu>
    )

    const MenuList = (props: MenuListProps<Option, false, GroupedOption>) => (
        <components.MenuList {...props} css={styles.menuList}>
            {props.children}
        </components.MenuList>
    )

    const Option = (props: OptionProps<Option, false, GroupedOption>) => (
        <components.Option {...props} css={[styles.option, props.isSelected && styles.optionSelected]}>
            {props.children}
        </components.Option>
    )

    const Placeholder = (props: PlaceholderProps<Option>) => (
        <components.Placeholder {...props} css={[styles.placeholder, props.selectProps.isDisabled && styles.disabled]}>
            {props.children}
        </components.Placeholder>
    )

    return (
        <Select
            {...rest}
            className={className}
            classNamePrefix='select'
            components={{ DropdownIndicator, Menu, MenuList, Option, SelectContainer, SingleValue, ValueContainer, Placeholder, Control }}
            css={[(theme) => styles.select(theme, size, disabled)]}
            defaultValue={defaultValue}
            isDisabled={disabled}
            isSearchable={isSearchable}
            menuIsOpen={menuIsOpen}
            name={name}
            onChange={onChange}
            options={options}
            ref={ref}
            styles={stylesOverride}
            value={value}
        />
    )
})

FormSelect.displayName = 'FormSelect'
FormSelect.defaultProps = defaultProps

export default FormSelect
