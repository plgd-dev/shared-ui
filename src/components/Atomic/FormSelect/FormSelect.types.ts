import { selectSizes, selectAligns } from './constants'

export type FormSelectSizeType = (typeof selectSizes)[keyof typeof selectSizes]
export type FormSelectAlignType = (typeof selectAligns)[keyof typeof selectAligns]

export type ValueType = string | number | boolean

export interface OptionType {
    readonly value: ValueType
    readonly label: string
}

export interface GroupedOption {
    readonly label: string
    readonly options: OptionType[]
}

export type Props = {
    align?: FormSelectAlignType
    autoWidth?: boolean
    checkboxOptions?: boolean
    className?: string
    creatable?: boolean
    dataTestId?: string
    defaultValue?: any
    disabled?: boolean
    error?: boolean
    footerLinksLeft?: { onClick: (values: any) => void; title: string; variant?: string }[]
    footerLinksRight?: { onClick: (values: any) => void; title: string; variant?: string }[]
    i18n?: {
        itemSelected: string
        itemsSelected: string
    }
    inlineStyle?: boolean
    isClearable?: boolean
    isMulti?: boolean
    isSearchable?: boolean
    menuIsOpen?: boolean
    menuPortalTarget?: HTMLElement | null
    menuZIndex?: number
    name?: string
    onBlur?: (v: any) => void
    onChange?: (v: any) => void
    onCreateOption?: (v: string | number) => void
    options: OptionType[]
    placeholder?: string
    size?: FormSelectSizeType
    value?: OptionType | OptionType[]
}

export const defaultProps: Partial<Props> = {
    align: selectAligns.LEFT,
    isMulti: false,
    size: selectSizes.NORMAL,
    placeholder: 'Select...',
}
