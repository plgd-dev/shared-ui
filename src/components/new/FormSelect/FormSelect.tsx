import { FC } from 'react'
import { GroupedOption, Option, Props, defaultProps } from './FormSelect.types'
import * as styles from './FormSelect.styles'
import Select, {
    components,
    ContainerProps,
    ControlProps,
    DropdownIndicatorProps,
    MenuListProps,
    MenuProps,
    OptionProps,
    PlaceholderProps,
    SingleValueProps,
    ValueContainerProps,
} from 'react-select'
import { Icon } from '../Icon'
import { selectSizes } from './constants'

const FormSelect: FC<Props> = (props) => {
    const { className, defaultValue, error, disabled, isSearchable, options, name, menuIsOpen, onChange, size, value } = { ...defaultProps, ...props }
    const stylesOverride = {
        menu: (base: any) => ({
            ...base,
            width: 'max-content',
            minWidth: '100%',
        }),
    }

    const Control = ({ children, ...props }: ControlProps<Option>) => (
        <components.Control {...props} css={[styles.control, error && styles.error, size === selectSizes.SMALL && styles.small]}>
            {children}
        </components.Control>
    )

    const ValueContainer = ({ children, ...props }: ValueContainerProps<Option>) => (
        <components.ValueContainer {...props} css={styles.valueContainer}>
            {children}
        </components.ValueContainer>
    )

    const SelectContainer = ({ children, ...props }: ContainerProps<Option>) => (
        <components.SelectContainer {...props} css={styles.selectContainer}>
            {children}
        </components.SelectContainer>
    )

    const SingleValue = ({ children, ...props }: SingleValueProps<Option>) => (
        <components.SingleValue {...props} css={[styles.value, props.selectProps.isDisabled && styles.disabled]}>
            {children}
        </components.SingleValue>
    )

    const DropdownIndicator = (props: DropdownIndicatorProps) => (
        <components.DropdownIndicator {...props} css={styles.dropdownIndicator}>
            <div css={[styles.indicator, props.selectProps.menuIsOpen && styles.indicatorOpen]}>
                <Icon icon='table-arrow-down' size={10} />
            </div>
        </components.DropdownIndicator>
    )

    const Menu = (props: MenuProps<Option, false, GroupedOption>) => (
        <components.Menu<Option, false, GroupedOption> {...props} css={styles.menu}>
            {props.children}
        </components.Menu>
    )

    const MenuList = (props: MenuListProps<Option, false, GroupedOption>) => (
        <components.MenuList {...props} css={styles.menuList}>
            {props.children}
        </components.MenuList>
    )

    const Option = (props: OptionProps<Option, false, GroupedOption>) => (
        <components.Option {...props} css={[styles.option, props.isSelected && styles.optionSelected]}>
            {props.children}
        </components.Option>
    )

    const Placeholder = (props: PlaceholderProps<Option>) => (
        <components.Placeholder {...props} css={[styles.placeholder, props.selectProps.isDisabled && styles.disabled]}>
            {props.children}
        </components.Placeholder>
    )

    return (
        <Select
            className={className}
            classNamePrefix='select'
            components={{ DropdownIndicator, Menu, MenuList, Option, SelectContainer, SingleValue, ValueContainer, Placeholder, Control }}
            css={styles.select(size, disabled)}
            defaultValue={defaultValue}
            isDisabled={disabled}
            isSearchable={isSearchable}
            menuIsOpen={menuIsOpen}
            name={name}
            onChange={onChange}
            options={options}
            styles={stylesOverride}
            value={value}
        />
    )
}

FormSelect.displayName = 'FormSelect'
FormSelect.defaultProps = defaultProps

export default FormSelect
