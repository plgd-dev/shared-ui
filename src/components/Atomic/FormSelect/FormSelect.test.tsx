import { render } from '../../../../test/jest-wrapper'
import FormSelect from './FormSelect'

const options = [
    { value: 'chocolate', label: 'Chocolate' },
    { value: 'strawberry', label: 'Strawberry' },
    { value: 'vanilla', label: 'Vanilla' },
]

describe('<FormSelect>', () => {
    it('render correctly - snapshot', () => {
        const { asFragment } = render(
            <div>
                <FormSelect name='form-select-1' options={options} />
                <br />
                <FormSelect defaultValue={options[1]} menuIsOpen={true} name='form-select-2' options={options} />
            </div>
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
