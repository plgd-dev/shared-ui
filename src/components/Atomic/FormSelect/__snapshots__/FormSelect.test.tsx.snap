// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSelect> render correctly - snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  box-sizing: border-box;
  width: 100%;
}

.emotion-0 .select__control .select__value-container .select__placeholder,
.emotion-0 .select__control .select__value-container .select__input-container {
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 22px;
  color: #252626!important;
}

.emotion-0 .select__control .select__value-container .select__placeholder input,
.emotion-0 .select__control .select__value-container .select__input-container input {
  font-family: 'Poppins',sans-serif;
  font-style: normal!important;
  font-weight: 400!important;
  font-size: 14px!important;
  line-height: 22px!important;
  color: #252626!important;
}

.emotion-0 .select__control.select__control--menu-is-open {
  border: 1px solid #007BBF;
}

.emotion-0 .select__control.select__control--is-focused {
  box-shadow: none;
}

.emotion-0 .select__indicator-separator {
  display: none;
}

.emotion-0 .select__input-container {
  margin: 0;
}

.emotion-1 {
  z-index: 9999;
  border: 0;
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  width: 1px;
  position: absolute;
  overflow: hidden;
  padding: 0;
  white-space: nowrap;
}

.emotion-3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: default;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  min-height: 38px;
  outline: 0!important;
  position: relative;
  -webkit-transition: all 100ms;
  transition: all 100ms;
  background-color: hsl(0, 0%, 100%);
  border-color: hsl(0, 0%, 80%);
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  box-sizing: border-box;
  height: 44px;
  min-height: 44px;
  border-radius: 8px;
  box-shadow: none;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  border: 1px solid #D7D8DA;
}

.emotion-3:hover {
  border-color: hsl(0, 0%, 70%);
}

.emotion-3:focus {
  outline: none;
}

.emotion-3:hover {
  outline: none;
  border: 1px solid #007BBF;
}

.emotion-4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: grid;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-overflow-scrolling: touch;
  position: relative;
  overflow: hidden;
  padding: 2px 8px;
  box-sizing: border-box;
  padding: 0 0 0 15px;
}

.emotion-5 {
  grid-area: 1/1/2/3;
  color: hsl(0, 0%, 50%);
  margin-left: 2px;
  margin-right: 2px;
  box-sizing: border-box;
  color: #252626;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 22px;
}

.emotion-6 {
  visibility: visible;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: inline-grid;
  grid-area: 1/1/2/3;
  grid-template-columns: 0 min-content;
  margin: 2px;
  padding-bottom: 2px;
  padding-top: 2px;
  color: hsl(0, 0%, 20%);
  box-sizing: border-box;
}

.emotion-6:after {
  content: attr(data-value) " ";
  visibility: hidden;
  white-space: pre;
  grid-area: 1/2;
  font: inherit;
  min-width: 2px;
  border: 0;
  margin: 0;
  outline: 0;
  padding: 0;
}

.emotion-7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
}

.emotion-8 {
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 1px;
  background-color: hsl(0, 0%, 80%);
  margin-bottom: 8px;
  margin-top: 8px;
  box-sizing: border-box;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: color 150ms;
  transition: color 150ms;
  color: hsl(0, 0%, 80%);
  padding: 8px;
  box-sizing: border-box;
  padding: 4px 16px 4px 4px;
}

.emotion-9:hover {
  color: hsl(0, 0%, 60%);
}

.emotion-10 {
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  color: #81868C;
}

.emotion-16 {
  grid-area: 1/1/2/3;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  color: hsl(0, 0%, 20%);
  margin-left: 2px;
  margin-right: 2px;
  box-sizing: border-box;
  color: #252626;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 22px;
  text-align: left;
}

.emotion-21 {
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  color: #81868C;
  -webkit-transform: rotate(180deg);
  -moz-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.emotion-22 {
  top: 100%;
  position: absolute;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  z-index: 1;
  background-color: hsl(0, 0%, 100%);
  border-radius: 4px;
  box-shadow: 0 0 0 1px hsla(0, 0%, 0%, 0.1),0 4px 11px hsla(0, 0%, 0%, 0.1);
  margin-bottom: 8px;
  margin-top: 8px;
  box-sizing: border-box;
  min-width: 100%;
  width: auto!important;
  margin-top: 4px;
  margin-bottom: 4px;
  border: 1px solid #E6E9ED;
  box-shadow: 0 30px 40px rgba(28, 52, 99, 0.1);
  border-radius: 8px;
  background: #FFFFFF;
  z-index: 1;
}

.emotion-23 {
  max-height: 300px;
  overflow-y: auto;
  position: relative;
  -webkit-overflow-scrolling: touch;
  padding-bottom: 4px;
  padding-top: 4px;
  box-sizing: border-box;
  text-align: left;
  padding: 8px 0;
}

.emotion-24 {
  cursor: default;
  display: block;
  font-size: inherit;
  width: 100%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  background-color: transparent;
  color: inherit;
  padding: 8px 12px;
  box-sizing: border-box;
  color: #81868C;
  background: #FFFFFF;
  font-style: normal;
  font-weight: 400;
  line-height: 22px;
  font-size: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  cursor: pointer;
  padding: 8px 16px;
}

.emotion-24:active {
  background-color: #B2D4FF;
}

.emotion-24:hover {
  color: #2261AE;
  background: transparent;
}

.emotion-24:active {
  background: transparent;
}

<div>
    <div
      class="emotion-0"
    >
      <span
        class="emotion-1"
        id="react-select-2-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="emotion-1"
        role="log"
      />
      <div
        class="select__control emotion-3"
      >
        <div
          class="select__value-container emotion-4"
        >
          <div
            class="select__placeholder emotion-5"
            id="react-select-2-placeholder"
          >
            Select...
          </div>
          <div
            class="select__input-container emotion-6"
            data-value=""
          >
            <input
              aria-activedescendant=""
              aria-autocomplete="list"
              aria-describedby="react-select-2-placeholder"
              aria-expanded="false"
              aria-haspopup="true"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              class="select__input"
              id="react-select-2-input"
              role="combobox"
              spellcheck="false"
              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="select__indicators emotion-7"
        >
          <span
            class="select__indicator-separator emotion-8"
          />
          <div
            aria-hidden="true"
            class="select__indicator select__dropdown-indicator emotion-9"
          >
            <div
              class="emotion-10"
            >
              <svg
                fill="none"
                height="10"
                viewBox="0 0 6 4"
                width="10"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m2.576 3.805-2.4-2.666a.72.72 0 0 1-.13-.727C.14.162.358 0 .601 0H5.4c.242 0 .461.162.555.412a.72.72 0 0 1-.13.727l-2.4 2.666a.56.56 0 0 1-.848 0"
                  fill="currentcolor"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <input
        name="form-select-1"
        type="hidden"
        value=""
      />
    </div>
    <br />
    <div
      class="emotion-0"
    >
      <span
        class="emotion-1"
        id="react-select-3-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="emotion-1"
        role="log"
      />
      <div
        class="select__control select__control--menu-is-open emotion-3"
      >
        <div
          class="select__value-container select__value-container--has-value emotion-4"
        >
          <div
            class="select__single-value emotion-16"
          >
            Strawberry
          </div>
          <div
            class="select__input-container emotion-6"
            data-value=""
          >
            <input
              aria-activedescendant=""
              aria-autocomplete="list"
              aria-controls="react-select-3-listbox"
              aria-expanded="true"
              aria-haspopup="true"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              class="select__input"
              id="react-select-3-input"
              role="combobox"
              spellcheck="false"
              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="select__indicators emotion-7"
        >
          <span
            class="select__indicator-separator emotion-8"
          />
          <div
            aria-hidden="true"
            class="select__indicator select__dropdown-indicator emotion-9"
          >
            <div
              class="emotion-21"
            >
              <svg
                fill="none"
                height="10"
                viewBox="0 0 6 4"
                width="10"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m2.576 3.805-2.4-2.666a.72.72 0 0 1-.13-.727C.14.162.358 0 .601 0H5.4c.242 0 .461.162.555.412a.72.72 0 0 1-.13.727l-2.4 2.666a.56.56 0 0 1-.848 0"
                  fill="currentcolor"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="select__menu emotion-22"
      >
        <div
          aria-multiselectable="false"
          class="select__menu-list emotion-23"
          id="react-select-3-listbox"
          role="listbox"
        >
          <div
            aria-disabled="false"
            aria-selected="false"
            class="select__option emotion-24"
            id="react-select-3-option-0"
            role="option"
            tabindex="-1"
          >
            Chocolate
          </div>
          <div
            aria-disabled="false"
            aria-selected="false"
            class="select__option emotion-24"
            id="react-select-3-option-2"
            role="option"
            tabindex="-1"
          >
            Vanilla
          </div>
        </div>
      </div>
      <input
        name="form-select-2"
        type="hidden"
        value="strawberry"
      />
    </div>
  </div>
</DocumentFragment>
`;
